test/index.test.js

```diff
@@ -694,10 +683,6 @@ test('Allow only request from with specific methods', t => {
   const fastify = Fastify()
   fastify.register(cors, { methods: ['GET', 'POST'] })
 
-  fastify.options('/', (req, reply) => {
-    reply.send('ok')
-  })
```

Comment:
why was this removed?

Reply:
The route never gets called by the test. I removed it to keep the tests concise. I only kept routes that actually get called. Do you prefer I put it back?

Reply:
I have a feeling this was called by the following `.inject()` call. why wasn't it called?

Reply:
You are correct, I confused this with the other tests.

Here, in the previous implementation, the request was handled and terminated in the onRequest hook since `preflightContinue` was the default value of false.

In my new implementation, when overriding an OPTIONS route, the preflight headers will not be added by the plugin (the `*` route handler will never execute). In order to test the `Access-Control-Allow-Methods` header, the route must _not_ be handled by a "user" route. Instead, it should be left to be handled by the route defined in this plugin.

Does that make more sense?

Reply:
Yes it does